### ISSUE TITLE
Windows PyPI package is missing 'dcm2bids'

### ISSUE THREAD

**Issue by kousu** [UNFmontreal/Dcm2Bids#79]
Currently, on Windows, if you do `pip install dcm2bids`, you end up with `C:\Users\$USER\AppData\Local\Programs\Python\Python38-32\Scripts\{dcm2bids,dcm2bids_scaffold,dcm2bids_helper,dcm2bids2.exe}`. Windows doesn't know about shebang lines, only about .exes, so only `dcm2bids2` runs, and it doesn't do anything

```
C:\Users\user\src>pip install dcm2bids
Collecting dcm2bids
  Using cached https://files.pythonhosted.org/packages/ab/b6/61a10d2f876ac78236d72a9aaa51a3b3f36f46220afb08cc57b280ee3511/dcm2bids-2.1.4-py2.py3-none-any.whl
Requirement already satisfied: future in c:\users\user\appdata\local\programs\python\python38-32\lib\site-packages (from dcm2bids) (0.18.2)
Installing collected packages: dcm2bids
Successfully installed dcm2bids-2.1.4
WARNING: You are using pip version 19.2.3, however version 20.1.1 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

C:\Users\user\src>Dcm2bids
'Dcm2bids' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\user\src>dir C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
 Volume in drive C has no label.
 Volume Serial Number is 1C55-94BA

 Directory of C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts

2020-05-26  03:05 AM    <DIR>          .
2020-05-26  03:05 AM    <DIR>          ..
2020-05-26  01:57 AM            93,068 chardetect.exe
2020-05-26  03:05 AM             3,340 dcm2bids
2020-05-26  03:05 AM            93,063 dcm2bids2.exe
2020-05-26  03:05 AM             1,242 dcm2bids_helper
2020-05-26  03:05 AM             2,884 dcm2bids_scaffold
2020-05-26  01:56 AM               449 dcmstack-script.py
2020-05-26  01:56 AM            65,536 dcmstack.exe
2020-05-26  01:56 AM               643 dcmstack.exe.manifest
2020-05-26  01:45 AM            93,077 easy_install-3.8.exe
2020-05-26  01:45 AM            93,077 easy_install.exe
2020-05-26  02:59 AM               446 futurize-script.py
2020-05-26  02:59 AM            65,536 futurize.exe
2020-05-26  02:59 AM               643 futurize.exe.manifest
2020-05-26  01:56 AM               445 nitool-script.py
2020-05-26  01:56 AM            65,536 nitool.exe
2020-05-26  01:56 AM               641 nitool.exe.manifest
2020-05-26  02:59 AM               450 pasteurize-script.py
2020-05-26  02:59 AM            65,536 pasteurize.exe
2020-05-26  02:59 AM               645 pasteurize.exe.manifest
2020-05-26  01:45 AM            93,059 pip.exe
2020-05-26  01:45 AM            93,059 pip3.8.exe
2020-05-26  01:45 AM            93,059 pip3.exe
              22 File(s)        925,434 bytes

```

```
C:\Users\user\src>Dcm2bids2
Traceback (most recent call last):
  File "c:\users\user\appdata\local\programs\python\python38-32\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\user\appdata\local\programs\python\python38-32\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts\dcm2bids2.exe\__main__.py", line 5, in <module>
ModuleNotFoundError: No module named 'dcm2bids.__main__'
```

If you unpack the [2.1.4 .whl](https://files.pythonhosted.org/packages/ab/b6/61a10d2f876ac78236d72a9aaa51a3b3f36f46220afb08cc57b280ee3511/dcm2bids-2.1.4-py2.py3-none-any.whl) you'll find

```
$ unzip dcm2bids-2.1.4-py2.py3-none-any.whl 
Archive:  dcm2bids-2.1.4-py2.py3-none-any.whl
  inflating: dcm2bids/__init__.py    
  inflating: dcm2bids/dcm2bids.py    
  inflating: dcm2bids/dcm2niix.py    
  inflating: dcm2bids/logger.py      
  inflating: dcm2bids/sidecar.py     
  inflating: dcm2bids/structure.py   
  inflating: dcm2bids/utils.py       
  inflating: dcm2bids/version.py     
  inflating: dcm2bids-2.1.4.data/scripts/dcm2bids  
  inflating: dcm2bids-2.1.4.data/scripts/dcm2bids_helper  
  inflating: dcm2bids-2.1.4.data/scripts/dcm2bids_scaffold  
  inflating: dcm2bids-2.1.4.dist-info/LICENSE.txt  
  inflating: dcm2bids-2.1.4.dist-info/METADATA  
  inflating: dcm2bids-2.1.4.dist-info/WHEEL  
  inflating: dcm2bids-2.1.4.dist-info/entry_points.txt  
  inflating: dcm2bids-2.1.4.dist-info/top_level.txt  
  inflating: dcm2bids-2.1.4.dist-info/RECORD  
```

and in particular this file:

```
$ cat dcm2bids-2.1.4.dist-info/entry_points.txt 
[console_scripts]
dcm2bids2 = dcm2bids.__main__:main
```

Which I suppose is an accidental result of this line at the time:

https://github.com/cbedetti/Dcm2Bids/blob/58e2ae280f2ad6ccb1bd4d6ceaa16ef9618815af/setup.py#L48

`dcm2bids2` doesn't run on Linux either:

```
$ pip install dcm2bids
Defaulting to user installation because normal site-packages is not writeable
WARNING: Keyring is skipped due to an exception: Failed to unlock the collection!
WARNING: Keyring is skipped due to an exception: Failed to unlock the collection!
Collecting dcm2bids
  WARNING: Keyring is skipped due to an exception: Failed to unlock the collection!
  Using cached dcm2bids-2.1.4-py2.py3-none-any.whl (28 kB)
Requirement already satisfied: future in /usr/lib/python3.8/site-packages (from dcm2bids) (0.18.2)
Installing collected packages: dcm2bids
Successfully installed dcm2bids-2.1.4
$ dcm2bids2
Traceback (most recent call last):
  File "/home/kousu/.local/bin/dcm2bids2", line 5, in <module>
    from dcm2bids.__main__ import main
ModuleNotFoundError: No module named 'dcm2bids.__main__'
```

but `dcm2bids` does -- on Linux:

```
$ dcm2bids
usage: dcm2bids [-h] -d DICOM_DIR [DICOM_DIR ...] -p PARTICIPANT [-s SESSION] -c CONFIG [-o OUTPUT_DIR] [--forceDcm2niix] [--clobber] [-l {DEBUG,INFO,WARNING,ERROR,CRITICAL}] [-a] [--version]
dcm2bids: error: the following arguments are required: -d/--dicom_dir, -p/--participant, -c/--config
```

but because of the missing `console_scripts` entry for it Windows' pip doesn't do the fixup .exe wrapper dance so this tool is unavailable to Windows users.

**Comment by kousu** [comment]
I just tried building my branch #80 on Windows and it produced a working install:

```
C:\Users\user\src\Dcm2bids>python setup.py install
running install
running bdist_egg
running egg_info
creating dcm2bids.egg-info
writing dcm2bids.egg-info\PKG-INFO
writing dependency_links to dcm2bids.egg-info\dependency_links.txt
writing entry points to dcm2bids.egg-info\entry_points.txt
writing requirements to dcm2bids.egg-info\requires.txt
writing top-level names to dcm2bids.egg-info\top_level.txt
writing manifest file 'dcm2bids.egg-info\SOURCES.txt'
reading manifest file 'dcm2bids.egg-info\SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'dcm2bids.egg-info\SOURCES.txt'
installing library code to build\bdist.win32\egg
running install_lib
running build_py
creating build
creating build\lib
creating build\lib\dcm2bids
copying dcm2bids\dcm2bids.py -> build\lib\dcm2bids
copying dcm2bids\dcm2niix.py -> build\lib\dcm2bids
copying dcm2bids\helper.py -> build\lib\dcm2bids
copying dcm2bids\logger.py -> build\lib\dcm2bids
copying dcm2bids\sidecar.py -> build\lib\dcm2bids
copying dcm2bids\structure.py -> build\lib\dcm2bids
copying dcm2bids\utils.py -> build\lib\dcm2bids
copying dcm2bids\version.py -> build\lib\dcm2bids
copying dcm2bids\__init__.py -> build\lib\dcm2bids
creating build\lib\tests
copying tests\test_dcm2bids.py -> build\lib\tests
copying tests\test_dcm2niix.py -> build\lib\tests
copying tests\test_sidecar.py -> build\lib\tests
copying tests\test_structure.py -> build\lib\tests
copying tests\test_version.py -> build\lib\tests
copying tests\__init__.py -> build\lib\tests
creating build\lib\dcm2bids\scaffold
copying dcm2bids\scaffold\__init__.py -> build\lib\dcm2bids\scaffold
creating build\bdist.win32
creating build\bdist.win32\egg
creating build\bdist.win32\egg\dcm2bids
copying build\lib\dcm2bids\dcm2bids.py -> build\bdist.win32\egg\dcm2bids
copying build\lib\dcm2bids\dcm2niix.py -> build\bdist.win32\egg\dcm2bids
copying build\lib\dcm2bids\helper.py -> build\bdist.win32\egg\dcm2bids
copying build\lib\dcm2bids\logger.py -> build\bdist.win32\egg\dcm2bids
creating build\bdist.win32\egg\dcm2bids\scaffold
copying build\lib\dcm2bids\scaffold\__init__.py -> build\bdist.win32\egg\dcm2bids\scaffold
copying build\lib\dcm2bids\sidecar.py -> build\bdist.win32\egg\dcm2bids
copying build\lib\dcm2bids\structure.py -> build\bdist.win32\egg\dcm2bids
copying build\lib\dcm2bids\utils.py -> build\bdist.win32\egg\dcm2bids
copying build\lib\dcm2bids\version.py -> build\bdist.win32\egg\dcm2bids
copying build\lib\dcm2bids\__init__.py -> build\bdist.win32\egg\dcm2bids
creating build\bdist.win32\egg\tests
copying build\lib\tests\test_dcm2bids.py -> build\bdist.win32\egg\tests
copying build\lib\tests\test_dcm2niix.py -> build\bdist.win32\egg\tests
copying build\lib\tests\test_sidecar.py -> build\bdist.win32\egg\tests
copying build\lib\tests\test_structure.py -> build\bdist.win32\egg\tests
copying build\lib\tests\test_version.py -> build\bdist.win32\egg\tests
copying build\lib\tests\__init__.py -> build\bdist.win32\egg\tests
byte-compiling build\bdist.win32\egg\dcm2bids\dcm2bids.py to dcm2bids.cpython-38.pyc
byte-compiling build\bdist.win32\egg\dcm2bids\dcm2niix.py to dcm2niix.cpython-38.pyc
byte-compiling build\bdist.win32\egg\dcm2bids\helper.py to helper.cpython-38.pyc
byte-compiling build\bdist.win32\egg\dcm2bids\logger.py to logger.cpython-38.pyc
byte-compiling build\bdist.win32\egg\dcm2bids\scaffold\__init__.py to __init__.cpython-38.pyc
byte-compiling build\bdist.win32\egg\dcm2bids\sidecar.py to sidecar.cpython-38.pyc
byte-compiling build\bdist.win32\egg\dcm2bids\structure.py to structure.cpython-38.pyc
byte-compiling build\bdist.win32\egg\dcm2bids\utils.py to utils.cpython-38.pyc
byte-compiling build\bdist.win32\egg\dcm2bids\version.py to version.cpython-38.pyc
byte-compiling build\bdist.win32\egg\dcm2bids\__init__.py to __init__.cpython-38.pyc
byte-compiling build\bdist.win32\egg\tests\test_dcm2bids.py to test_dcm2bids.cpython-38.pyc
byte-compiling build\bdist.win32\egg\tests\test_dcm2niix.py to test_dcm2niix.cpython-38.pyc
byte-compiling build\bdist.win32\egg\tests\test_sidecar.py to test_sidecar.cpython-38.pyc
byte-compiling build\bdist.win32\egg\tests\test_structure.py to test_structure.cpython-38.pyc
byte-compiling build\bdist.win32\egg\tests\test_version.py to test_version.cpython-38.pyc
byte-compiling build\bdist.win32\egg\tests\__init__.py to __init__.cpython-38.pyc
creating build\bdist.win32\egg\EGG-INFO
copying dcm2bids.egg-info\PKG-INFO -> build\bdist.win32\egg\EGG-INFO
copying dcm2bids.egg-info\SOURCES.txt -> build\bdist.win32\egg\EGG-INFO
copying dcm2bids.egg-info\dependency_links.txt -> build\bdist.win32\egg\EGG-INFO
copying dcm2bids.egg-info\entry_points.txt -> build\bdist.win32\egg\EGG-INFO
copying dcm2bids.egg-info\requires.txt -> build\bdist.win32\egg\EGG-INFO
copying dcm2bids.egg-info\top_level.txt -> build\bdist.win32\egg\EGG-INFO
zip_safe flag not set; analyzing archive contents...
dcm2bids.scaffold.__pycache__.__init__.cpython-38: module references __file__
tests.__pycache__.test_dcm2bids.cpython-38: module references __file__
tests.__pycache__.test_dcm2niix.cpython-38: module references __file__
tests.__pycache__.test_sidecar.cpython-38: module references __file__
creating dist
creating 'dist\dcm2bids-2.1.4-py3.8.egg' and adding 'build\bdist.win32\egg' to it
removing 'build\bdist.win32\egg' (and everything under it)
Processing dcm2bids-2.1.4-py3.8.egg
creating c:\users\user\appdata\local\programs\python\python38-32\lib\site-packages\dcm2bids-2.1.4-py3.8.egg
Extracting dcm2bids-2.1.4-py3.8.egg to c:\users\user\appdata\local\programs\python\python38-32\lib\site-packages
Adding dcm2bids 2.1.4 to easy-install.pth file
Installing dcm2bids-script.py script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing dcm2bids.exe script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing dcm2bids.exe.manifest script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing dcm2bids_helper-script.py script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing dcm2bids_helper.exe script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing dcm2bids_helper.exe.manifest script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing dcm2bids_scaffold-script.py script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing dcm2bids_scaffold.exe script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing dcm2bids_scaffold.exe.manifest script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts

Installed c:\users\user\appdata\local\programs\python\python38-32\lib\site-packages\dcm2bids-2.1.4-py3.8.egg
Processing dependencies for dcm2bids==2.1.4
Searching for future==0.18.2
Best match: future 0.18.2
Adding future 0.18.2 to easy-install.pth file
Installing futurize-script.py script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing futurize.exe script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing futurize.exe.manifest script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing pasteurize-script.py script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing pasteurize.exe script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts
Installing pasteurize.exe.manifest script to C:\Users\user\AppData\Local\Programs\Python\Python38-32\Scripts

Using c:\users\user\appdata\local\programs\python\python38-32\lib\site-packages
Finished processing dependencies for dcm2bids==2.1.4

C:\Users\user\src\Dcm2bids>dcm2bids
usage: dcm2bids [-h] -d DICOM_DIR [DICOM_DIR ...] -p PARTICIPANT [-s SESSION] -c CONFIG [-o OUTPUT_DIR] [--forceDcm2niix] [--clobber]
                [-l {DEBUG,INFO,WARNING,ERROR,CRITICAL}] [-a]
dcm2bids: error: the following arguments are required: -d/--dicom_dir, -p/--participant, -c/--config
```

So I think this bug is already fixed. All you need to do is republish to PyPI and this bug will go away.

**Comment by kousu** [comment]
And if you accept #81, republishing to PyPI will stop being a pain :)

**Comment by arnaudbore** [comment]
Fixed 2.1.6rc1

**Comment by kousu** [comment]
Actually, this was fixed by https://github.com/UNFmontreal/Dcm2Bids/commit/134199b4a9f9776fc7fea271e089d83a39828295. It was even working in 2.1.5:

![image](https://user-images.githubusercontent.com/987487/108096068-8a996200-704e-11eb-8981-23a5aa3d05a5.png)


 I don't know what PR that commit was a part of! Too bad, because the only comment says "Update setup.py" without explaining what was going on. I don't think this bug was fixed intentionally, but I'm glad it's fixed :)
